jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/Fabric8HttpUtilTest.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import org.junit.jupiter.api.Test;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class Fabric8HttpUtilTest {
+
+  @Test
+  void toFormData_whenDataProvidedAsMap_thenCreateFormDataPayload() throws UnsupportedEncodingException {
+    // Given
+    Map<String, String> formDataMap = new HashMap<>();
+    formDataMap.put("grant_type", "password");
+    formDataMap.put("refresh_token", "secret");
+    formDataMap.put("service", "auth.example.com");
+    formDataMap.put("scope", "repository:myuser/test-chart:pull");
+    formDataMap.put("client_id", "EclipseJKube");
+    formDataMap.put("username", "myuser");
+    formDataMap.put("password", "secret");
+
+    // When
+    String formDataPayload = Fabric8HttpUtil.toFormData(formDataMap);
+
+    // Then
+    assertThat(formDataPayload)
+        .isEqualTo("refresh_token=secret&password=secret&grant_type=password&service=auth.example.com&scope=repository%3Amyuser%2Ftest-chart%3Apull&client_id=EclipseJKube&username=myuser");
+  }
+
+  @Test
+  void extractAuthenticationChallengeIntoMap_whenWwwHeaderProvided_thenParseDataIntoMap() {
+    // Given
+    String wwwAuthenticateValue = "Bearer realm=\"https://auth.example.com/token\",service=\"registry.example.com\",scope=\"repository:myuser/test-chart:pull\"";
+    Map<String, List<String>> responseHeaders = new HashMap<>();
+    responseHeaders.put("WWW-Authenticate", Collections.singletonList(wwwAuthenticateValue));
+    HttpResponse<byte[]> response = new TestFabric8HttpResponse(HTTP_OK, responseHeaders, null, null);
```

Comment:
We might want to see if we can publish and reuse the Fabric8 Client classes we've got for this.